addons/mrp_repair/models/mrp_repair.py

```diff
@@ -0,0 +1,539 @@
+    operations_ids = fields.One2many('mrp.repair.line', 'repair_id', string='Operation Lines', readonly=True, states={'draft': [('readonly', False)]}, copy=True, oldname="operations")
+    pricelist_id = fields.Many2one('product.pricelist', string='Pricelist', default=lambda self: self.env['product.pricelist'].search([('type', '=', 'sale')], limit=1), help='Pricelist of the selected partner.')
+    partner_invoice_id = fields.Many2one('res.partner', string='Invoicing Address')
+    invoice_method = fields.Selection([("none", "No Invoice"), ("b4repair", "Before Repair"), ("after_repair", "After Repair")], string="Invoice Method", index=True, required=True, default=lambda *a: 'none', states={'draft': [('readonly', False)]}, readonly=True, help='Selecting \'Before Repair\' or \'After Repair\' will allow you to generate invoice before or after the repair is done respectively. \'No invoice\' means you don\'t want to generate invoice for this repair order.')
+    invoice_id = fields.Many2one('account.invoice', string='Invoice', readonly=True, track_visibility="onchange", copy=False)
+    move_id = fields.Many2one('stock.move', string='Move', readonly=True, help="Move created by the repair order", track_visibility="onchange", copy=False)
+    fees_lines_ids = fields.One2many('mrp.repair.fee', 'repair_id', string='Fees', readonly=True, states={'draft': [('readonly', False)]}, copy=True, oldname="fees_lines")
+    internal_notes = fields.Text(string='Internal Notes')
+    quotation_notes = fields.Text(string='Quotation Notes')
+    company_id = fields.Many2one('res.company', string='Company', default=lambda self: self.env['res.company']._company_default_get('mrp.repair'))
+    invoiced = fields.Boolean(string='Invoiced', readonly=True, copy=False)
+    repaired = fields.Boolean(string='Repaired', readonly=True, copy=False)
+    amount_untaxed = fields.Float(string='Untaxed Amount', compute='_amount_untaxed', store=True)
+    amount_tax = fields.Float(string='Taxes', compute='_amount_tax', store=True)
+    amount_total = fields.Float(string='Total', compute='_amount_total', store=True)
+
+    _sql_constraints = [('name', 'unique (name)', 'The name of the Repair Order must be unique!'), ]
+
+    @api.onchange('product_id')
+    def onchange_product_id(self):
+        """ On change of product sets some values.
+        @param product_id: Changed product
+        @return: Dictionary of values.
+        """
+        if self.product_id:
+            self.guarantee_limit = False
+            self.lot_id = False
+            self.product_uom_id = self.product_id.uom_id.id
+
+    @api.onchange('product_uom_id')
+    def onchange_product_uom_id(self):
+        res = []
+        if not self.product_uom_id or not self.product_id:
+            return res
+        if self.product_uom_id.category_id.id != self.product_id.uom_id.category_id.id:
+            raise UserError('The Product Unit of Measure you chose has a different category than in the product form.')
```

Comment:
In onchange method you have to return warning dictionary like:  {'warning': {'title': 'Warning', 'message': 'Message'}}.